agent/agent.go

```diff
@@ -156,6 +156,10 @@ LOOP:
 				grip.Notice("Next task response indicates agent should exit")
 				return nil
 			}
+			// if the host's current task group is finished we teardown
+			if nextTask.ShouldTeardownGroup {
+				needPostGroup = nextTask.ShouldTeardownGroup
+			}
```

Comment:
isn't this redundant and you could just loose the if?

Reply:
I didn't do this originally but lint complains--needPostGroup could be set from a previous iteration in the loop, and we wouldn't want to reset it to False if ShouldTeardownGroup isn't set.